themes/theme-gmd/index.jsx

```diff
@@ -11,12 +11,19 @@ import React from 'react';
 import { render } from 'react-dom';
 import { configureStore } from '@shopgate/pwa-common/store';
 import { appWillStart } from '@shopgate/pwa-common/action-creators/app';
+import { i18n } from '@shopgate/engage/core';
 import smoothscroll from 'smoothscroll-polyfill';
 import fetchClientInformation from '@shopgate/pwa-common/actions/client/fetchClientInformation';
+import locales from './locale';
 import reducers from './pages/reducers';
 import subscribers from './pages/subscribers';
 import Pages from './pages';
 
+i18n.init({
+  locales,
+  lang: process.env.LOCALE,
+});
+
```

Comment:
Since i18n module is ready, you can pass it as a prop to <Pages i18n={i18n}> and forward it to <App i18n={i18n}>

Reply:
I'd rather import it within the I18nProvider and deprecate current props it's using.

Reply:
Implemented new one everywhere I think + added deprecation warning if some externals would use I18nProvider directly.

All should be fully backwards compatible. Also added some cases to the demo extension.